FrontEnd/Web_Vitals.md

```diff
@@ -0,0 +1,346 @@
+```
+
+After:
+
+```css
+background: linear-gradient(135deg, #fbc6ff 20%, #bdfff9 90%);
+```
+
+
+**이미지 및 레이아웃 이동**
+
+브라우저는 이미지가 로드된 후에 이미지의 크기를 결정할 수 있습니다. 페이지가 렌더링된 후 이미지 로드가 발생하지만 이미지에 예약된 공간이 없는 경우 이미지가 나타날 때 레이아웃 이동이 발생합니다. 데모페이지에서 히어로 이미지는 로드 시 레이아웃 이동을 유발합니다.
+
+> 이미지가 느리게 연결되거나 파일 크기가 특히 큰 이미지를 로드하는 등 이미지 로드 속도가 느린 상황에서 레이아웃 이동을 일으키는 현상은 더욱 뚜렷합니다.
+
+
+
+**이미지 레이아웃 이동 문제**
+
+명시적 너비와 높이가 없는 이미지를 확인하려면 Lighthouse의 "이미지 요소에는 명시적 너비와 높이가 있습니다" 감사를 사용하십시오.
+
+![image](https://user-images.githubusercontent.com/53992007/131064544-b603c164-284c-4d01-8571-83e3d14cdb38.png)
+
+
+이 예제에서는 히어로 이미지와 아티클 이미지 모두에 너비 및 높이 속성이 누락되었습니다.
+
+**위 사례를 수정하는 방법**
+
+레이아웃 이동을 방지하기 위해 이 이미지에 너비 및 높이 속성을 설정합니다.
+
+
+
+Before:
+
+```css
+<img src="https://source.unsplash.com/random/2000x600" alt="image to load in">
+<img src="https://source.unsplash.com/random/800x600" alt="image to load in">
+```
+
+After:
+
+```css
+<img src="https://source.unsplash.com/random/2000x600" width="2000" height="600" alt="image to load in">
+<img src="https://source.unsplash.com/random/800x600" width="800" height="600" alt="image to load in">
+```
+
+
+
+> 이미지를 로드 하는 또 다른 방법은 너비 및 속성을 지정하는 것과 함께, srcset 및 크기 속성을 사용하는 것입니다. 이렇게 하면 크기가 다른 이미지를 다른 장치에 제공할 수 있다는 추가적인 성능 이점이 있습니다.
+
+
+
+#### 글꼴
+
+글꼴은 텍스트 렌더링을 지연시키고 레이아웃 이동을 유발할 수 있습니다. 따라서 글꼴을 빨리 전달하는 것이 중요합니다.
+
+**지연된 텍스트 렌더링**
+
+기본적으로 연결된 웹 글꼴이 아직 로드되지 않은 경우 브라우저는 텍스트 요소를 즉시 렌더링하지 않습니다. 이는 "스타일 없는 텍스트의 플래시"(FOUT)를 방지하기 위해 수행됩니다. 대부분의 경우 이 경우 FCP(First Contentful Paint)가 지연됩니다. 경우에 따라 최대 내용물 페인트(LCP)가 지연됩니다.
+
+기본적으로 크롬 기반 및 Firefox 브라우저는 연결된 웹 글꼴이 로드되지 않은 경우 최대 3초 동안 텍스트 렌더링을 차단하고 Safari는 텍스트 렌더링을 무기한 차단합니다. 차단 기간은 브라우저가 웹 글꼴을 요청할 때 시작됩니다. 차단 기간이 끝날 때까지 글꼴이 로드되지 않은 경우 브라우저는 fallback 글꼴을 사용하여 텍스트를 렌더링하고 웹 글꼴에서 스왑합니다.
+
+
+**레이아웃 이동**
+
+
+글꼴 스와핑은 사용자에게 콘텐츠를 빠르게 표시하는 데 탁월하지만 레이아웃 이동을 유발할 수 있습니다. 이러한 레이아웃 이동은 웹 글꼴과 fallback 글꼴이 페이지에서 다른 공간을 차지할 때 발생합니다. 유사한 비율의 글꼴을 사용하면 이러한 레이아웃 이동의 크기가 최소화됩니다.
+
+
+![image](https://user-images.githubusercontent.com/53992007/131064984-7e057597-6a5a-45bd-a156-6baef3e49e0f.png)
+
+
+**글꼴 관련 문제**
+
+
+특정 페이지에서 로드 중인 글꼴을 보려면 개발자 도구에서 네트워크 탭을 열고 글꼴별로 필터링합니다. 글꼴은 큰 파일이 될 수 있으므로 일반적으로 적은 수의 글꼴만 사용하는 것이 성능에 더 좋습니다.
+
+
+![image](https://user-images.githubusercontent.com/53992007/131069982-c92e0292-31fb-443e-9a6e-39918f602735.png)
+
+글꼴을 요청하는 데 걸리는 시간을 보려면 [타이밍] 탭을 클릭합니다. 글꼴을 빨리 요청할수록 더 빨리 로드하여 사용할 수 있습니다.
+
+![image](https://user-images.githubusercontent.com/53992007/131070122-e3a72005-5034-468e-9363-5e21b58edd6a.png)
+
+
+
+
+**위 사례를 수정하는 방법**
+
+이 예제에서는 Google 글꼴 API를 사용합니다. Google 글꼴은 `<link>` 태그 또는 @import 문을 통해 글꼴을 로드하는 옵션을 제공합니다. `<link>` 코드에는 `사전 연결` 리소스 힌트가 포함되어 있습니다. 따라서 @import 버전을 사용하는 것보다 스타일시트가 더 빨리 전달됩니다.
+
+매우 높은 수준에서 리소스 힌트를 브라우저에 `특정 연결`을 설정하거나 `특정 리소스`를 다운로드해야 함을 암시하는 방법으로 생각할 수 있습니다. 따라서 브라우저는 이러한 작업의 우선 순위를 지정합니다. 리소스 힌트를 사용할 때 특정 작업의 우선 순위를 지정하게되면 브라우저 리소스가 다른 작업(우선하지 않은 작업)에서 제거됩니다. 따라서 리소스 힌트는 주의하여 사용해야 합니다. 
```

Comment:
'매우 높은 수준에서' 라는 말이 높은 리소스 우선순위를 뜻하는 건가요? (이건 제가 잘 모르겠어서... 혹시 조금만 더 풀어서 설명해주실 수 있을까요)

Reply:
이 부분은 
At a very high-level, you can think of resource hints as a way to hint to the browser that it will need to set up a particular connection or download a particular resource.

이 문장인데 at a very high-level이라는 부분을 굳이 번역하지 않고 빼도 상관없을 것 같아요 그래서 빼겠습니다